test/e2e/shared/suite.go

```diff
@@ -182,7 +182,8 @@ func AllNodesBeforeSuite(e2eCtx *E2EContext, data []byte) {
 	e2eCtx.Settings.ArtifactFolder = conf.ArtifactFolder
 	e2eCtx.Settings.ConfigPath = conf.ConfigPath
 	e2eCtx.Environment.ClusterctlConfigPath = conf.ClusterctlConfigPath
-	e2eCtx.Environment.BootstrapClusterProxy = framework.NewClusterProxy("bootstrap", conf.KubeconfigPath, e2eCtx.Environment.Scheme)
+	withLogCollector := framework.WithMachineLogCollector(OpenStackLogCollector{E2EContext: *e2eCtx})
+	e2eCtx.Environment.BootstrapClusterProxy = framework.NewClusterProxy("bootstrap", conf.KubeconfigPath, e2eCtx.Environment.Scheme, withLogCollector)
```

Comment:
This is where the magic happens. We pass the log collector to the cluster proxy so the CAPI test specs can make use of it. The only exception I found is the clusterctl upgrade test. There is doesn't help that the bootstrap cluster proxy has the log collector, because it is the workload cluster that is of interest. I'm planning to work on a way to pass the log collector also to the workload cluster in CAPI to improve this in the future.